java/ql/src/experimental/Security/CWE/CWE-691/InsecureClassLoading.java

```diff
@@ -0,0 +1,30 @@
+
+import android.app.Application;
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.os.Bundle;
+
+public class InsecureClassLoading extends Application {
+  @Override
+  public void onCreate() {
+    super.onCreate();
+    invokePlugins();
+  }
+
+  private void invokePlugins() {
+    for (PackageInfo info : getPackageManager().getInstalledPackages(0)) {
```

Comment:
The badness here is somewhat subtle, so suggest adding a comment -- the badness is that getInstalledPackages(...) enumerates packages belonging to other applications, so they could have dropped their own package that appears to be one of our plugins, right?

Reply:
yes, I have added a comment in the code to reflect the same.